src/lib/core/overlay/position/connected-position-strategy.ts

```diff
@@ -89,11 +95,23 @@ export class ConnectedPositionStrategy implements PositionStrategy {
   }
 
   /** Sets the layout direction so the overlay's position can be adjusted to match. */
-  setDirection(dir: 'ltr' | 'rtl') {
+  withDirection(dir: 'ltr' | 'rtl'): ConnectedPositionStrategy {
```

Comment:
The return type should actually just be `this` (which is a cool but little-known TS feature)